### ISSUE TITLE
fabric8 menu - manage connections to multiple openshift + kubernetes clusters

### ISSUE THREAD

**Issue by rawlingsj** [fabric8io/fabric8#6778]
We could really do with a dev experience to help work with multiple clusters from our laptops.
Ideally a menu that can be installed onto OSX, Windows, Linux which manages connections to remote kubernetes + openshift clusters and provide an easy switch of namespaces.

Personally I have 4 terminals right now that are connected to different clusters so I can perform `kubectl` / `oc` interactions,  without the manual set up of the back ground colour I'd be totally lost and I've lost count of the number times I've almost deleted our CD infra namespace on Google Container Engine.

Currently my workflow when I want to connect to a new cluster is in a new terminal window:

```
export ~/kube/conf-$CLUSTER_NAME
```
then either 
```
oc login foo
```
 or 
```
kubectl config set-credentials bar --username=admin --password=secret
kubectl config set-cluster bar --insecure-skip-tls-verify=true --server=https://10.n.n.n
kubectl config set-context bar --user=bar --namespace=cd --cluster=bar
kubectl config use-context bar
```
this stores the auth details in the new `~/kube/conf-$CLUSTER_NAME` file so I don't overwrite the remote connection the next time I log into a different cluster or use a differnt shell.

__What we could do with__
What would be great is an app I can install onto my laptop which looks at my current `~/kube/conf` and extracts the existing known clusters and creates a separate `~/kube/conf-$CLUSTER_NAME`  file which can then be used in the same way as above.

Next open a new terminal window and create a connection, change the background (somehow associate profiles to clusters so the same colours are used after a restart) and label the window with the cluster name + current namespace?

The app in my top menu can then create new connections to new clusters avoiding the horrible `kubectl` commands described above.  It could then also discover namespaces so a user could easily switch between environments.  In openshift thats just `oc project cheese` but its more complicated with plain kubernetes.

A few months back I started looking at https://github.com/alexflint/gallium which we could use to build desktop applications in Go and HTML.  I started playing around with it but didn't get that far https://github.com/rawlingsj/fabric8menu.  I'm sure there's others out there and I'm not fussy what's used if we can get a maintainable open source app that can be installed on OSX, Windows and Linux (preferably in that order)

**Comment by jstrachan** [comment]
I've found myself building shells scripts to setup terminals to point to different known clusters which are of the form:
```
#!/bin/sh
export NAME=foo
echo "using the $NAME cluster"
mkdir -p ~/.kube/${NAME}
export KUBECONFIG=~/.kube/${NAME}/config
cd ~/.kube/${NAME}
bash
```
Then setting up separate iTerm profiles (with different titles/backgrounds) so to switch between shells without worry of accidentally deleting stuff in production ;) - I just switch terminal windows.

Then the OOTB kubectl I reserve for minikube/minishift (which I don't mind trashing now and again when I'm about to re-install mini*).

But it'd be really nice to have a UI that wrapped up this stuff; so that you could easily add a new cluster; it'd make a new .kube config dir and setup the UI stuff to open a pre-configured shell for it there.

**Comment by rawlingsj** [comment]
yeah that's pretty much the same as me so I'm sure everyone that works with openshift or kubernetes is going to need this!  

for example this is a screenshot of the four clusters I'm always using throughout my day.. having a way to automate connecting and setting this up is sorely needed.

![screenshot 2017-03-24 09 23 43](https://cloud.githubusercontent.com/assets/1457180/24288034/c5d795ce-1073-11e7-9f72-ecd52e804d81.png)
